regulation-worker/internal/delete/api/api.go

```diff
@@ -109,6 +110,22 @@ func (api *APIManager) deleteWithRetry(ctx context.Context, job model.Job, desti
 	jobStatus := getJobStatus(resp.StatusCode, jobResp)
 	pkgLogger.Debugf("[%v] Job: %v, JobStatus: %v", destination.Name, job.ID, jobStatus)
 
+	if isOAuthEnabled && isAuthStatusInactiveErrorCategory(jobResp) {
+		dest := &backendconfig.DestinationT{
+			ID:     destination.DestinationID,
+			Config: destination.Config,
+			DestinationDefinition: backendconfig.DestinationDefinitionT{
+				Name:   destination.Name,
+				Config: destination.DestDefConfig,
+			},
+		}
+		stCode, response := api.OAuth.UpdateAuthStatusToInactive(dest, job.WorkspaceID, oAuthDetail.id)
+		jobStatus.Status = model.JobStatusAborted
```

Comment:
Shouldn't this be `JobStatusUndefined` if `stCode != http.StatusOK`?

Reply:
What does `JobStatusUndefined` signify in terms of regulation functionality ?